src/main/java/com/Model/Patient.java

```diff
@@ -13,20 +14,23 @@
 public class Patient {
 
     @Id
-    @GeneratedValue (strategy = GenerationType.AUTO)
-    private long Id;
+    private String Id;
 
-    @OneToMany
+    @OneToMany (cascade = CascadeType.ALL)
     private Set <PatientEventAssociation> patientEventAssociations;
 
     //All patients are anonymous, name is not needed for constructor
     @Autowired
     public Patient(){
```

Comment:
how is the id of the patient set if the default constructor is called?

Reply:
I will generate a UUID inside the default constructor.